### PR TITLE
feat: allow disabling ssl verification for project for git clone

### PR DESCRIPTION
**Pull request by kurokobo**
Related to: #317 

This PR makes following changes with unit/integration tests:

- Add `verify_ssl` (boolean, `True` by default) field to `Project` model (b9790bb060d5002c00cf7d0bcfd27d75d77ae99d)
- For `git` subprocess for `clone`, set `GIT_SSL_NO_VERIFY` environment variable to `true` if `verify_ssl` is `False` for the project (133a00c5d77bedc6fb167ec1e1603841e9bcff39)

If this PR is acceptable, I will work on adding checkbox to the UI to toggle this field in handy way.

This is the first time for me to send PR for this project, so please let me know if I'm on the wrong way.
I would be happy to have your review, but if for some reason it is difficult for you to proceed with this PR, please do not hesitate to close it. Thanks in advance!

## Additional information

All tests have been passed as follows:

```bash
$ task test
task: [test] poetry run python -m pytest 
============================================= test session starts ==============================================
platform linux -- Python 3.9.7, pytest-7.3.1, pluggy-1.0.0
django: settings: aap_eda.settings.development (from env)
rootdir: /home/kuro/work/kurokobo/eda-server
configfile: pytest.ini
plugins: django-4.5.2, asyncio-0.21.0
asyncio: mode=auto
collected 192 items                                                                                            

tests/integration/api/test_activation.py ...............                                                 [  7%]
tests/integration/api/test_activation_instance.py .......                                                [ 11%]
tests/integration/api/test_auth.py ....                                                                  [ 13%]
tests/integration/api/test_awx_token.py .......                                                          [ 17%]
tests/integration/api/test_credential.py ......                                                          [ 20%]
tests/integration/api/test_decision_environment.py ......                                                [ 23%]
tests/integration/api/test_extra_var.py ....                                                             [ 25%]
tests/integration/api/test_pagination.py ....                                                            [ 27%]
tests/integration/api/test_permissions.py ....                                                           [ 29%]
tests/integration/api/test_playbook.py .....                                                             [ 32%]
tests/integration/api/test_project.py ..................                                                 [ 41%]
tests/integration/api/test_rulebook.py ..................................                                [ 59%]
tests/integration/api/test_self.py .                                                                     [ 59%]
tests/integration/api/test_task.py ...                                                                   [ 61%]
tests/integration/api/test_user.py ...........                                                           [ 67%]
tests/integration/services/test_activation_db_logger.py ..                                               [ 68%]
tests/integration/services/test_activation_podman.py ......                                              [ 71%]
tests/integration/services/test_project_import.py .....                                                  [ 73%]
tests/integration/services/test_ruleset_activate.py ......s                                              [ 77%]
tests/integration/tasks/test_ruleset.py .s                                                               [ 78%]
tests/unit/test_git.py .                                                                                 [ 79%]
tests/integration/api/test_activation.py ....                                                            [ 81%]
tests/integration/api/test_awx_token.py .                                                                [ 81%]
tests/integration/wsapi/test_consumer.py .........                                                       [ 86%]
tests/integration/api/test_auth.py .                                                                     [ 86%]
tests/integration/core/test_views.py .                                                                   [ 87%]
tests/unit/test_ansible_rulebook.py ..                                                                   [ 88%]
tests/unit/test_encryption.py ..                                                                         [ 89%]
tests/unit/test_git.py ......                                                                            [ 92%]
tests/unit/test_permissions.py ..............                                                            [100%]
```

Also tested with `curl` with following cases:

- Case 1: Add new project without `verify_ssl` -> Sync failed (Expected)
- Case 2: Update existing project with `verify_ssl: false` -> Sync completed
- Case 3: Add new project with `verify_ssl: false` -> Sync completed

### Case 1: Add new project without `verify_ssl` -> Sync failed (Expected)

Note: `https://git.example.com` is my private git server with self-signed certificate.

```bash
# Create new project without specifying verify_ssl.
# Project is created with verify_ssl = true by default
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "POST" \
>   -H "Content-Type: application/json" \
>   -d '{"name":"demo","url":"https://git.example.com/gitea/demo.git"}' \     👈👈👈
>   'https://eda.example.com/api/eda/v1/projects/' | jq
{
  "name": "demo",
  "description": "",
  "credential_id": null,
  "verify_ssl": true,     👈👈👈
  "id": 3,
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "",
  "import_state": "pending",
  "import_error": null,
  ...
}

# Sync failed as expected since ssl verification is enabled for this project
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "GET" \
>   "https://eda.example.com/api/eda/v1/projects/3/" | jq
{
  "id": 3,
  "name": "demo",
  "description": "",
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "",
  "verify_ssl": true,
  "credential": null,
  "import_state": "failed",     👈👈👈
  "import_error": "Command '/usr/bin/git clone --quiet --depth 1 https://git.example.com/gitea/demo.git /tmp/eda-project-el4xh9l_/src' returned non-zero exit status 128.",     👈👈👈
  ...
}

# Worker shows the cause of the error is self-signed cert
$ kubectl -n eda logs deployment/eda-default-worker
...
2023-08-13 12:19:01,812 INFO     Task started: Import project ( project_id=3 )
2023-08-13 12:19:02,099 WARNING  Command returned non-zero exit status 128:
        command: /usr/bin/git clone --quiet --depth 1 https://git.example.com/gitea/demo.git /tmp/eda-project-el4xh9l_/src
        stderr: fatal: unable to access 'https://git.example.com/gitea/demo.git/': SSL certificate problem: self-signed certificate     👈👈👈
...
```

### Case 2: Update existing project with `verify_ssl: false` -> Sync completed

```bash
# Patch the project that created above with verify_ssl = false
# The project is updated with verify_ssl = False
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "PATCH" \
>   -H "Content-Type: application/json" \
>   -d '{"verify_ssl":false}' \     👈👈👈
>   'https://eda.example.com/api/eda/v1/projects/3/' | jq
{
  "name": "demo",
  "description": "",
  "credential_id": null,
  "verify_ssl": false,     👈👈👈
  "id": 3,
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "",
  "import_state": "failed",
  "import_error": "Command '/usr/bin/git clone --quiet --depth 1 https://git.example.com/gitea/demo.git /tmp/eda-project-el4xh9l_/src' returned non-zero exit status 128.",
  ...
}

# Request project sync
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "POST" \
>   -H "Content-Type: application/json" \
>   'https://eda.example.com/api/eda/v1/projects/3/sync/' | jq
{
  "name": "demo",
  "description": "",
  "credential_id": null,
  "verify_ssl": false,
  "id": 3,
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "",
  "import_state": "pending",
  "import_error": null,
  ...
}

# Sync completed as expected
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "GET" \
>   "https://eda.example.com/api/eda/v1/projects/3/" | jq
{
  "id": 3,
  "name": "demo",
  "description": "",
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "512c5c5a0452b474825c90aa3787963cff981e62",
  "verify_ssl": false,
  "credential": null,
  "import_state": "completed",     👈👈👈
  "import_error": null,     👈👈👈
  ...
}
```

### Case 3: Add new project with `verify_ssl: false` -> Sync completed

```bash
# Create new project with verify_ssl = false
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "POST" \
>   -H "Content-Type: application/json" \
>   -d '{"name":"ssl_verification_disabled","url":"https://git.example.com/gitea/demo.git","verify_ssl":false}' \     👈👈👈
>   'https://eda.example.com/api/eda/v1/projects/' | jq
{
  "name": "ssl_verification_disabled",
  "description": "",
  "credential_id": null,
  "verify_ssl": false,     👈👈👈
  "id": 4,
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "",
  "import_state": "pending",
  "import_error": null,
  ...
}

# Sync completed
$ curl -ksu "${BASIC_AUTH_SECRET}" \
>   -X "GET" \
>   "https://eda.example.com/api/eda/v1/projects/4/" | jq
{
  "id": 4,
  "name": "ssl_verification_disabled",
  "description": "",
  "url": "https://git.example.com/gitea/demo.git",
  "git_hash": "512c5c5a0452b474825c90aa3787963cff981e62",
  "verify_ssl": false,
  "credential": null,
  "import_state": "completed",     👈👈👈
  "import_error": null,     👈👈👈
  ...
}
```

### CODE REVIEW

**Comment by kurokobo** [comment]
I have also created a PR to ansible-ui that allows this field to be configured via the Web UI: https://github.com/ansible/ansible-ui/pull/783

Since I don't know if this PR will be approved, I mark that PR as a draft at this time.

image

**Comment by mkanoor** [comment]
@kurokobo Can you please rebase this PR, it has migration conflicts with some of the other PR's that got merged

**Comment by kurokobo** [comment]
@mkanoor 
Thanks for moving this forward 😃 
I've update my branch and confirm that all test have been passed.

File: src/aap_eda/services/project/git.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ def clone(
         *,
         credential: Optional[Credential] = None,
         depth: Optional[int] = None,
+        verify_ssl: Optional[bool] = True,
```
Comment (Alex-Izquierdo) [inline src/aap_eda/services/project/git.py:102]: Type hint would be incorrect, this param should be just a boolean. 
```suggestion
        verify_ssl: bool = True,
```

**Comment by kurokobo** [comment]
@Alex-Izquierdo 
Thanks for reviewing this! I've rebased with the latest `main` branch and fixed my branch.

**Review by Alex-Izquierdo — DISMISSED** [review]

**Comment by kurokobo** [comment]
@Alex-Izquierdo 
Thanks for approving this, the check was failing due to conflicts in migration scripts, so I've re-created the migration file. Sorry for dismissing your approval by my update😢

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by hsong-rh — APPROVED** [review]

**Pull request merged**